### ISSUE TITLE
Android Application

### ISSUE THREAD

**Issue by alyhasansakr** [EdgeLab-FHDO/Latency-Measurement#3]
We need to develop a simple Android application that detects the position of a marker.
The marker may be just a single color circle, or a simple 2-D code.

**Comment by alyhasansakr** [comment]
First we need to choose which IDE we should use, my recommendation is Unity Game Engine. It is easy to use and we can use it afterwards for AR devices without much modification.

**Comment by alyhasansakr** [comment]
We need both a front-end and a back-end, the front-end runs locally on the Android device, and the back-end can run either locally on the Android device or remotely on a Linux server.

**Comment by Satyam06** [comment]
> First we need to choose which IDE we should use, my recommendation is Unity Game Engine. It is easy to use and we can use it afterwards for AR devices without much modification.

How do I get access to it, and would we require the AR device while executing/validating the IDE.?

**Comment by alyhasansakr** [comment]
You can get the student version from https://unity.com/solutions/game
As a first step we are going to run it only on Android.
You may need to learn a lot of new stuff, I can help you with any of the steps.

**Comment by alyhasansakr** [comment]
We want something similar to this https://assetstore.unity.com/packages/templates/tutorials/markerbased-ar-example-29678

**Comment by Satyam06** [comment]
Will start with this right away, will let you know the status of it.

**Comment by Satyam06** [comment]
[05/08, 9:04 pm] Satyam: I am going through the codes, I was trying the same in Jupyter

[05/08, 9:06 pm] Aly: ok, good we can continue this week and maybe the next one as well only with OpenCV, but after that you need to know some basics about Unity.

[05/08, 9:08 pm] Satyam: So is it okay to try with python or should we use the similar framework that is provided on the unity page

[06/08, 12:43 am] Aly: unity is its own platform, so I would prefer that you use it right now, however it may be too much to learn at one moment, if you want to start with python that is ok, but only until end of next week then we must switch to unity or find another platform

**Comment by Satyam06** [comment]
1. To check the possibility of running the developed openCV python codes on android..?
2. For Unity setup, the .py code must be converted to c#
3. Generate the code for circle detection/localization.

**Comment by alyhasansakr** [comment]
Just finished with the professor, we focus only on reading a code, we don't need the circle or face detection for now..

**Comment by Satyam06** [comment]
1. Alright, so we read a pattern or the Aruco code..?
2. The color detection.py is what I had started with in the unity, currently.
3. Also this is feasible on android using tensorflow, we can discuss it further.

**Comment by alyhasansakr** [comment]
1. We read Aruco code.
1. Could you make a branch with your code and make a pull-request so that I can see what exactly you are doing. Let me know if you need help with that.
1. I need to investigate more about Android, will come back to you about that.

**Comment by Satyam06** [comment]
Yeah, I will share the code as a branch. Will let you know if i need any help.
Will wait for your response on the android, will continue with UNITY till then.

**Comment by alyhasansakr** [comment]
I have merged the pull-request, now we need to have this on Unity.

**Comment by Satyam06** [comment]
Tensorflow application development using the developed codes for platform independent latency measurement.
